Cython/Compiler/ExprNodes.py

```diff
@@ -4749,9 +4749,29 @@ def analyse_types(self, env, getting=True):
                 analyse_base=False)
 
         if self.start:
-            self.start = self.start.analyse_types(env)
+            self.start = CondExprNode(
+                self.start.pos,
+                true_val = IntNode(self.start.pos, value = '0'),
+                false_val = self.start,
+                test = PrimaryCmpNode(
+                    self.start.pos,
+                    operand1 = self.start,
```

Comment:
If `self.start`is non-trivial, reusing the node in more than one place won't work. You have to wrap it in a `ResultRefNode` and the expression during which it will be used in an `EvalWithTempExprNode`. You can find several examples of that in `Optimize.py`.

That also suggests that we should have a test for a slice value that is e.g. a function call. There is probably already a test for that, which also checks for side-effects (to prevent duplicate evaluation).

Reply:
Added - I'm not seeing any test errors caused by my lack of use of `EvalWithTempExprNode` / `ResultRefNode` logic, at least from what I can tell. I'll go ahead and add one.